167 General asset type info

najimgh opened the pull request:
Modified all UI views and Controllers to add a System Type Menu button. Linked appropriate buttons between the System Type views. Modified Views to reflect the mock-ups on confluence.

najimgh committed:
Issue (#167)
- Added/Modified pages for the system type info and system type edit to match the mock-ups on confluence.
- Added the System Type menu button to all of the FXML and linked.
- Linked Buttons on each page to the appropriate FXML.

najimgh committed:
Issue (#167)
- Added System Type Image to be displayed dynamically depending on the system type.
- Added Model tab to the System Edit Page.

jeffbuonamici commented:
Am I supposed to be seeing something here?

![image](https://user-images.githubusercontent.com/11137499/105128951-0e911600-5ab2-11eb-8445-7eae04768d20.png)

jeffbuonamici commented on JavaFXUserInterface/src/main/java/SystemTypeEditController.java:
```diff
@@ -0,0 +1,95 @@
+     *
+     * @author Najim
+     */
+    public void setImage() {
+        if (systemTypeName.getText().equals("Engine")) {
+            systemTypeImageView.setImage(new Image("imgs/system_type_engine.png"));
+        } else {
+            systemTypeImageView.setImage(new Image("imgs/system_type_engine.png"));
+        }
+    }
```
This isn't dynamic. If someone enters a new Asset Type, this won't be able to match an image for that new type.

najimgh replied:
Then we will probably need to implement a feature where the user can choose an image for a system type. Otherwise, I don't see how I could make this dynamic, especially if the user wants to enter a new Asset Type.

jeffbuonamici commented on JavaFXUserInterface/src/main/java/SystemTypeInfoController.java:
```diff
@@ -0,0 +1,108 @@
+     *
+     * @author Najim
+     */
+    public void setImage() {
+        if (systemTypeName.getText().equals("Engine")) {
+            systemTypeImageView.setImage(new Image("imgs/system_type_engine.png"));
+        } else {
+            systemTypeImageView.setImage(new Image("imgs/system_type_engine.png"));
+        }
+    }
```
Same dynamic issue.

najimgh replied:
Same reply as above

jeffbuonamici commented on JavaFXUserInterface/src/main/resources/SystemTypeEdit.fxml:
```diff
@@ -53,69 +70,162 @@
+                                    <TextField fx:id="thresholdOK" layoutX="220.0" layoutY="101.0" prefHeight="25.0"
+                                               prefWidth="350.0"/>
+                                    <Text fill="#0c072e" layoutX="82.0" layoutY="119.0" strokeType="OUTSIDE"
+                                          strokeWidth="0.0" text="Threshold: OK">
+                                        <font>
+                                            <Font name="Segoe UI Bold" size="14.0"/>
+                                        </font>
+                                    </Text>
+                                    <TextField fx:id="thresholdAdvisory" layoutX="220.0" layoutY="141.0"
+                                               prefHeight="25.0" prefWidth="350.0"/>
+                                    <Text fill="#0c072e" layoutX="82.0" layoutY="159.0" strokeType="OUTSIDE"
+                                          strokeWidth="0.0" text="Threshold: Advisory">
+                                        <font>
+                                            <Font name="Segoe UI Bold" size="14.0"/>
+                                        </font>
+                                    </Text>
+                                    <TextField fx:id="thresholdCaution" layoutX="220.0" layoutY="181.0"
+                                               prefHeight="25.0" prefWidth="350.0"/>
+                                    <Text fill="#0c072e" layoutX="82.0" layoutY="199.0" strokeType="OUTSIDE"
+                                          strokeWidth="0.0" text="Threshold: Caution">
+                                        <font>
+                                            <Font name="Segoe UI Bold" size="14.0"/>
+                                        </font>
+                                    </Text>
+                                    <TextField fx:id="thresholdWarning" layoutX="220.0" layoutY="221.0"
+                                               prefHeight="25.0" prefWidth="350.0"/>
+                                    <Text fill="#0c072e" layoutX="82.0" layoutY="239.0" strokeType="OUTSIDE"
+                                          strokeWidth="0.0" text="Threshold: Warning">
+                                        <font>
+                                            <Font name="Segoe UI Bold" size="14.0"/>
+                                        </font>
+                                    </Text>
+                                    <TextField fx:id="thresholdFailed" layoutX="220.0" layoutY="261.0" prefHeight="25.0"
+                                               prefWidth="350.0"/>
+                                    <Text fill="#0c072e" layoutX="82.0" layoutY="279.0" strokeType="OUTSIDE"
+                                          strokeWidth="0.0" text="Threshold: Failed">
+                                        <font>
+                                            <Font name="Segoe UI Bold" size="14.0"/>
+                                        </font>
+                                    </Text>
```
This may have the dynamic issue as well. Might be misremembering what the PO said.

najimgh replied:
Paul told me to put 5 thresholds for now with the names OK, Advisory, Caution, Warning and Failed.

PaulMicu replied:
there might have been a miss-understanding, I did not say to hardcode the javafx elements with the 5 thresholds given by the po

najimgh replied:
After further discussion, kept the hardcoded thresholds and removed the model field.

jeffbuonamici commented on JavaFXUserInterface/src/main/resources/SystemTypeInfo.fxml:
```diff
@@ -46,67 +62,171 @@
+                                        <TextField fx:id="thresholdOK" layoutX="220.0" layoutY="101.0" prefHeight="25.0"
+                                                   prefWidth="350.0"/>
+                                        <Text fill="#0c072e" layoutX="82.0" layoutY="119.0" strokeType="OUTSIDE"
+                                              strokeWidth="0.0" text="Threshold: OK">
+                                            <font>
+                                                <Font name="Segoe UI Bold" size="14.0"/>
+                                            </font>
+                                        </Text>
+                                        <TextField fx:id="thresholdAdvisory" layoutX="220.0" layoutY="141.0"
+                                                   prefHeight="25.0" prefWidth="350.0"/>
+                                        <Text fill="#0c072e" layoutX="82.0" layoutY="159.0" strokeType="OUTSIDE"
+                                              strokeWidth="0.0" text="Threshold: Advisory">
+                                            <font>
+                                                <Font name="Segoe UI Bold" size="14.0"/>
+                                            </font>
+                                        </Text>
+                                        <TextField fx:id="thresholdCaution" layoutX="220.0" layoutY="181.0"
+                                                   prefHeight="25.0" prefWidth="350.0"/>
+                                        <Text fill="#0c072e" layoutX="82.0" layoutY="199.0" strokeType="OUTSIDE"
+                                              strokeWidth="0.0" text="Threshold: Caution">
+                                            <font>
+                                                <Font name="Segoe UI Bold" size="14.0"/>
+                                            </font>
+                                        </Text>
+                                        <TextField fx:id="thresholdWarning" layoutX="220.0" layoutY="221.0"
+                                                   prefHeight="25.0" prefWidth="350.0"/>
+                                        <Text fill="#0c072e" layoutX="82.0" layoutY="239.0" strokeType="OUTSIDE"
+                                              strokeWidth="0.0" text="Threshold: Warning">
+                                            <font>
+                                                <Font name="Segoe UI Bold" size="14.0"/>
+                                            </font>
+                                        </Text>
+                                        <TextField fx:id="thresholdFailed" layoutX="220.0" layoutY="261.0"
+                                                   prefHeight="25.0" prefWidth="350.0"/>
+                                        <Text fill="#0c072e" layoutX="82.0" layoutY="279.0" strokeType="OUTSIDE"
+                                              strokeWidth="0.0" text="Threshold: Failed">
+                                            <font>
+                                                <Font name="Segoe UI Bold" size="14.0"/>
+                                            </font>
+                                        </Text>
```
Same thing.

najimgh replied:
Same reply as above

najimgh replied:
After further discussion, kept the hardcoded thresholds and removed the model field.

jeffbuonamici reviewed (changes requested):
Description text boxes need to wrap text properly. Look at Add System page for reference.

najimgh commented:
There is an issue for populating the system types page (Issue: #156 ) Shirwa is working on it.

najimgh committed:
Issue (#167)
- Added an attribute on the system type description text box in the FXML to wrap text appropriately.

jeffbuonamici commented:
Ok. My concerns have been addressed, thanks.

PaulMicu commented on JavaFXUserInterface/src/main/java/SystemTypeEditController.java:
```diff
@@ -0,0 +1,95 @@
+                uiUtilities.changeScene(mouseEvent, "/SystemTypeList");
+            }
+        });
+//        //Attach link to infoEditBtn to go to SystemTypeEdit.fxml
+//        infoSaveBtn.setOnMouseClicked(new EventHandler<MouseEvent>() {
+//            @Override
+//            public void handle(MouseEvent mouseEvent) {
+//                uiUtilities.changeScene(mouseEvent, "/SystemTypeEdit");
+//            }
+//        });
```
don't commit this type of comment

najimgh replied:
removed

PaulMicu commented on JavaFXUserInterface/src/main/java/SystemTypeEditController.java:
```diff
@@ -0,0 +1,95 @@
+                uiUtilities.changeScene(mouseEvent, "/SystemTypeInfo");
+            }
+        });
+//        //Attach function to deleteBtn to delete an asset type
+//        infoDeleteBtn.setOnMouseClicked(new EventHandler<MouseEvent>() {
+//            @Override
+//            public void handle(MouseEvent mouseEvent) {}
+//        });
+//        //Attach function to deleteBtn to delete an asset type
+//        modelDeleteBtn.setOnMouseClicked(new EventHandler<MouseEvent>() {
+//            @Override
+//            public void handle(MouseEvent mouseEvent) {}
+//        });
```
don't commit this type of comment

najimgh replied:
removed

PaulMicu reviewed (changes requested):
please see comments

JayChouteau commented on JavaFXUserInterface/src/main/java/AddSystemTypeController.java:
```diff
@@ -72,6 +74,13 @@ public void handle(MouseEvent mouseEvent) {
             }
         });
+        //Attach link to systemTypeMenuBtn to go to SystemTypeList.fxml
+        systemTypeMenuBtn.setOnMouseClicked(new EventHandler<MouseEvent>() {
```
All of the mouse events that are causing the scenes to be changed can be transformed into lambda expressions. This was done as part of the SonarQube refactoring for code smells. When you over the event, Intellij suggests to you. Formatting change. I will only note this change request here, but it would need to be done on all similar events that you are pushing

najimgh replied:
All similar mouseEvent have been refactored into a lambda expression

JayChouteau reviewed (changes requested):
Some change requests when it comes to formatting.

najimgh committed:
Issue (#167)
- Removed unecessary comments
- Refactored MouseEvent into lambda expressions
- Changed the order of fields in the system type info and system type edit
- Removed the model field in the system type info and system type edit

najimgh committed:
Merge branch 'develop' into 167_general_asset_type_info

najimgh commented:
Merge Conflicts Resolved

najimgh committed:
Merge branch 'develop' of https://github.com/rdsaliba/notorious-eng into 167_general_asset_type_info

 Conflicts:
	JavaFXUserInterface/src/main/java/SystemTypeController.java
	JavaFXUserInterface/src/main/resources/SystemTypeList.fxml

najimgh committed:
Issue (#167)
- Linked the System type list page to the System type info page
- Added functionality to pass data between the system type list, system type info and system type edit pages
- Modified the way the system type image is set

najimgh commented:
All issues mentioned above have been resolved accordingly. Linked the system type list with system type info and system type edit.

najimgh committed:
Issue (#167)
- Fixed List View Empty Cell for the system type list
- Fixed the positioning of menu buttons in the system type list view

PaulMicu commented on JavaFXUserInterface/src/main/java/SystemsController.java:
```diff
@@ -64,7 +66,7 @@ public SystemsController() {
 
         try {
-            systems = FXCollections.observableArrayList(ModelController.getInstance().getAllLiveAssets());
+            systems = FXCollections.observableArrayList(ModelController.getInstance().getAllLiveAssets().subList(0, 50));
```
remove this committed line

najimgh replied:
Removed

PaulMicu reviewed (changes requested):
one small change see comment

najimgh committed:
Issue (#167)
- Removed sublist of systems
